os/drivers/lcd/mipi_lcd.c

```diff
@@ -174,7 +243,13 @@ static int lcd_putarea(FAR struct lcd_dev_s *dev, fb_coord_t row_start, fb_coord
 	row_end += 1;
 	col_start += 1;
 	col_end += 1;
+#if defined(CONFIG_LCD_ST7701SN) && (defined(CONFIG_LCD_LANDSCAPE) || defined(CONFIG_LCD_RLANDSCAPE))
```

Comment:
It doesn't look good to have lcd specific code here.

How about adding a new config like CONFIG_LCD_SW_ROTATION?